// react/jsx-dev-runtime.js
(function (global, factory) {
  typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports) :
  typeof define === 'function' && define.amd ? define(['exports'], factory) :
  (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory((global["jsx-dev-runtime"] = global["jsx-dev-runtime"] || {}, global["jsx-dev-runtime"].js = {})));
})(this, (function (exports) { 'use strict';

  var supportSymbol = typeof Symbol === 'function' && Symbol["for"];
  var REACT_ELEMENT_TYPE = supportSymbol
      ? Symbol["for"]('react.element')
      : 0xeac7;

  // 创建ReactElement
  var ReactElement = function (type, key, ref, props) {
      var element = {
          $$typeof: REACT_ELEMENT_TYPE,
          type: type,
          key: key,
          ref: ref,
          props: props,
          __mark: 'custom'
      };
      return element;
  };
  var jsx = function (type, config) {
      var maybeChildren = [];
      for (var _i = 2; _i < arguments.length; _i++) {
          maybeChildren[_i - 2] = arguments[_i];
      }
      var key = null;
      var props = {};
      var ref = null;
      // 遍历config
      // 把遍历到的prop 赋值给props
      for (var prop in config) {
          var val = config[prop];
          // 如果是 key
          // 那么转化成string
          if ('key' === prop && undefined !== val) {
              key = '' + val;
              continue;
          }
          // 如果是ref 直接赋值给ref
          if ('ref' === prop && undefined !== val) {
              ref = val;
              continue;
          }
          if ({}.hasOwnProperty.call(config, prop)) {
              props[prop] = val;
          }
      }
      // 获取children
      var maybeChildrenLength = maybeChildren.length;
      if (maybeChildrenLength) {
          if (1 === maybeChildrenLength) {
              props.children = maybeChildren[0];
          }
          else {
              props.children = maybeChildren;
          }
      }
      // 最终返回创建的element
      return ReactElement(type, key, ref, props);
  };
  var jsxDEV = function (type, config) {
      var key = null;
      var props = {};
      var ref = null;
      // 遍历config
      // 把遍历到的prop 赋值给props
      for (var prop in config) {
          var val = config[prop];
          // 如果是 key
          // 那么转化成string
          if ('key' === prop && undefined !== val) {
              key = '' + val;
              continue;
          }
          // 如果是ref 直接赋值给ref
          if ('ref' === prop && undefined !== val) {
              ref = val;
              continue;
          }
          if ({}.hasOwnProperty.call(config, prop)) {
              props[prop] = val;
          }
      }
      // 最终返回创建的element
      return ReactElement(type, key, ref, props);
  };

  exports.jsx = jsx;
  exports.jsxDEV = jsxDEV;

}));
